projects/packages/stats/src/class-tracking-pixel.php

```diff
@@ -0,0 +1,173 @@
+		if ( $wp_the_query->is_single || $wp_the_query->is_page || $wp_the_query->is_posts_page ) {
+			// Store and reset the queried_object and queried_object_id
+			// Otherwise, redirect_canonical() will redirect to home_url( '/' ) for show_on_front = page sites where home_url() is not all lowercase.
+			// Repro:
+			// 1. Set home_url = https://ExamPle.com/
+			// 2. Set show_on_front = page
+			// 3. Set page_on_front = something
+			// 4. Visit https://example.com/ !
+			$queried_object    = isset( $wp_the_query->queried_object ) ? $wp_the_query->queried_object : null;
+			$queried_object_id = isset( $wp_the_query->queried_object_id ) ? $wp_the_query->queried_object_id : null;
+			try {
+				$post_obj = $wp_the_query->get_queried_object();
+				$post     = $post_obj instanceof WP_Post ? $post_obj->ID : '0';
+			} finally {
+				$wp_the_query->queried_object    = $queried_object;
+				$wp_the_query->queried_object_id = $queried_object_id;
+			}
+		} else {
+			$post = '0';
+		}
+
+		return compact( 'v', 'j', 'blog', 'post', 'tz', 'srv' );
+	}
+
+	/**
+	 * Stats Footer.
+	 *
+	 * @access public
+	 * @return void
+	 */
+	public static function add_to_footer() {
+		$data = self::build_view_data();
+		// TODO Is this really needed here?
+		if ( class_exists( 'Jetpack_AMP_Support' ) && \Jetpack_AMP_Support::is_amp_request() ) {
+
+			/**
+			 * Filter the parameters added to the AMP pixel tracking code.
+			 *
+			 * @module stats
+			 *
+			 * @since-jetpack 10.9
+			 *
+			 * @param array $data Array of options about the site and page you're on.
+			 */
+			$data = (array) apply_filters( 'jetpack_stats_footer_amp_data', $data );
+			self::render_amp_footer( $data );
+		} else {
+
+			/**
+			 * Filter the parameters added to the JavaScript stats tracking code.
+			 *
+			 * @module stats
+			 *
+			 * @since-jetpack 10.9
+			 *
+			 * @param array $data Array of options about the site and page you're on.
+			 */
+			$data = (array) apply_filters( 'jetpack_stats_footer_js_data', $data );
+			self::render_footer( $data );
+		}
+
+	}
+
+	/**
+	 * Render the stats footer
+	 *
+	 * @access public
+	 * @param array $data Array of data for the JS stats tracker.
+	 */
+	public static function render_footer( $data ) {
+		// phpcs:disable WordPress.WP.EnqueuedResources.NonEnqueuedScript
+		// When there is a way to use defer with enqueue, we can move to it and inline the custom data.
+		$script           = 'https://stats.wp.com/e-' . gmdate( 'YW' ) . '.js';
+		$data_stats_array = self::stats_array( $data );
+
+		$stats_footer = <<<END
+	<script src='{$script}' defer></script>
+	<script>
+		_stq = window._stq || [];
+		_stq.push([ 'view', {{$data_stats_array}} ]);
+		_stq.push([ 'clickTrackerInit', '{$data['blog']}', '{$data['post']}' ]);
+	</script>
+END;
+		// phpcs:enable
+		print $stats_footer; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
Is there a reason why the rendering approach in `render_amp_footer` is different?

Reply:
Regarding if it makes sense to do the `get::footer `method, for me it makes total sense and was my initial approach. That way the testing is way easier. But then we discussed that it might be better to keep them for backward compatibility. I am not sure how to check if those methods are being used or not directly, aside from the stats module itself.

Reply:
oh, right. in this case, we could leave the rendering methods as is but extract the composition logic into separate methods